lib/parser/parser_stream.js

```diff
@@ -28,12 +29,16 @@ inherits(ParserStream, WritableStream);
 
 //WritableStream implementation
 ParserStream.prototype._write = function (chunk, encoding, callback) {
+    this.written = true;
     this.writeCallback = callback;
     this.parser.tokenizer.write(chunk.toString('utf8'), this.lastChunkWritten);
     this._runParsingLoop();
 };
 
 ParserStream.prototype.end = function (chunk, encoding, callback) {
+    if (!this.written && typeof chunk === 'undefined')
```

Comment:
Btw, `chunk === undefined` would be equivalent and simpler. Or you can even do `chunk = chunk || ''`.